modules/gis/src/lib/geojson-to-binary.js

```diff
@@ -73,87 +68,133 @@ function firstPass(features) {
 
           // eslint-disable-next-line max-depth
           for (const coord of flatten(polygon)) {
-            // eslint-disable-next-line max-depth
-            if (coord.length === 3) maxCoordLength = 3;
+            coordLengths.add(coord.length);
           }
         }
         break;
       default:
         throw new Error(`Unsupported geometry type: ${geometry.type}`);
     }
+
+    if (feature.properties) {
+      for (const key in feature.properties) {
+        const val = feature.properties[key];
+        numericProps[key] = numericProps[key] ? isNumeric(val) : numericProps[key];
+      }
+    }
   }
 
   return {
     pointPositions,
     linePositions,
     linePaths,
-    coordLength: maxCoordLength,
+    coordLength: Math.max(...coordLengths),
     polygonPositions,
     polygonObjects,
-    polygonRings
+    polygonRings,
+    // Array of keys whose values are always numeric
+    numericProps: Object.keys(numericProps).filter(k => numericProps[k])
   };
 }
 
 // Second scan over GeoJSON features
 // Fills coordinates into pre-allocated typed arrays
-function secondPass(features, firstPassData, options = {}) {
+function secondPass(features, options = {}) {
   const {
     pointPositions,
```

Comment:
Idea: should we call these `pointPositionCount` etc? As I read this I keep asking myself if this is an array of positions or just the count.